packages/headless/src/controllers/result-list/headless-result-list.ts

```diff
@@ -51,6 +51,31 @@ export function buildResultList(
     dispatch(registerFieldsToInclude(options.fieldsToInclude));
   }
 
+  let lastFetchCompleted = 0;
+  let consecutiveFetches = 0;
+  const maxConsecutiveFetches = 3;
+  const maxDelayBetweenFetches = 200;
+
+  const triggerFetchMoreResult = () => {
```

Comment:
Basically checks that fetchMoreResult if not called with little to no delay in between requests too often.

Reply:
If it is called without delay too often, it won't dispatch and log an error to the console.

Reply:
It's not supposed to kick in when using Atomic currently, manually scrolling, or even using the keyboard `end` button should go under 200ms most times.

It kicks in when constantly triggering fetchMoreResult, like having a call on the lifecycle hook "componentDidRender" for example.

Basically what it does here is:
Once the fetchMoreResult dispatch is completed, if a new dispatch is executed in less than 200ms, increment the counter. If the dispatch delay is over 200ms, reset it.
Once the counter reaches 5, meaning 5 fetchMoreResult requests have been executed with little to no delay between them. It likely means that somewhere in the user's code there is some kind of "infinite loop/request" going on.
When that happens, we log an error and return, to get out of the "infinite request situation".

Reply:
> It kicks in when constantly triggering fetchMoreResult, like having a call on the lifecycle hook "componentDidRender" for example.

Wouldn't this be prevented by the `isLoading` guard though? If you add a for loop in Atomic that tries to call the function 100 times, does the error log?

> When that happens, we log an error and return, to get out of the "infinite request situation".

The early return will prevent the dispatch which is good. However, wouldn't we send another 5 requests just 200 ms later?

Reply:
> Wouldn't this be prevented by the isLoading guard though? If you add a for loop in Atomic that tries to call the function 100 times, does the error log?

The isLoading would prevent you from calling very very quickly inside a loop, true. 

What would not be prevented though is if your "infinite scrolling container" inside your app, because of CSS rules and what not, can never be filled, and will always require more results. 

Does that scenario make sense ?

To put into contrast, here is a `valid` scenario that we would not want to block: 

* A single user load a search page.
* An infinite scrolling mechanism in the englobing app start requesting more result to fill an infinite scrolling list.
* Once 10 results returns, because the end user has a very big screen, the app need to request 10 more to "fill up the screen" (or, the scrolling container).
* Eventually, after a couple of successive request, the scrolling container for our imaginary end user is "full", and the end users sees a full web page, and the requests stop.
* The above scenario could be optimized by the englobing app by requesting more results per page (50 instead of 10, for example). But, nothing forces the app to do this optimization.


And here is an `invalid` scenario we want to prevent:

* A single user load a search page.
* An infinite scrolling mechanism in the englobing app start requesting more result to fill an infinite scrolling list.
* Once 10 results returns, let's assume there's a bug in the app.
* The bug in the app makes it so it "always think it should request more result to fill the scroll container", in order to display a full web page of results to en end user. AKA, it never stops, as opposed to the valid scenario above.
* After 100 requests are done, since we now reached result # 1000, the API starts returning errors, that will spam non stop until the end user leaves the page.
* Multiply that by thousand of users per day on a website == "the index is starting to catch on fire, QPM are through the roof" etc.

You will see in the second scenario, the requests are not "as fast as they can be inside a loop", they are actually waiting correctly on each successive request.

It's just that they never stop.

Reply:
As for the arbitrary delay and number of retry... it's indeed very arbitrary. It's essentially a "guess" that something is wrong.

I don't have any good suggestion out of the top of my head to make it less arbitrary, though.

Reply:
A bit arbitrary but it's realistic after doing some tests.
With current implementation, it will allow the first scenario (up to 5 times) and block the second scenario.

Added a variable in order not to log a bunch of errors, and updated the lastFetchCompleted variable when returning early